cli/internal/cmd/upgradeapply.go

```diff
@@ -172,11 +176,18 @@ func (u *upgradeApplyCmd) upgradeApply(cmd *cobra.Command, upgradeDir string, fl
 		return err
 	}
 
+	// TODO(msanft): Remove reading from idFile once v2.12.0 is released and read from state file directly.
+	// For now, this is only here to ensure upgradability from an id-file to a state file version.
 	var idFile clusterid.File
 	if err := u.fileHandler.ReadJSON(constants.ClusterIDsFilename, &idFile); err != nil {
 		return fmt.Errorf("reading cluster ID file: %w", err)
 	}
```

Comment:
We should add some additional logic here in case a state file already exists.
E.g. try to read from state file first, and in case of file-not-found fall back to reading ID file